ui/app/services/permissions.js

```diff
@@ -86,11 +86,11 @@ export default Service.extend({
     this.set('canViewAll', null);
   },
 
-  hasNavPermission(navItem, routeParams) {
+  hasNavPermission(navItem, routeParams, capabilities = [null]) {
```

Comment:
Had to double think this again then remembered seeing it before somewhere:

https://github.com/hashicorp/vault/pull/6094#discussion_r251556511

Not now obvs, but I wonder if it's worth at some point changing the `hasPermission` method to just check for a zero length array and returning false if so. I dunno the `[null]` thing always jumps out as slightly weird looking. I dunno, what do you think?

You potentially don't need to add your default value here aswell as in `hasPermission` also as you are just passing `capabilities` through and therefore defaulting twice. Again, one of those 'I could go one way or the other' depending on things, thought I'd mention though.

Reply:
i looked into removing the default value here but we still need it since `navPathParams` calls `hasPermission` without any capabilities. 

similarly, we don't want to return false when no capabilities have been specified because we still want to see whether the user has access to the path overall (meaning we check for any capability other than deny). this happens inside `hasMatchingExactPath` and `hasMatchingGlobPath`.

great tips though, i appreciate your eye for clean code! :D